src/main/java/io/tus/java/client/TusUploader.java

```diff
@@ -159,6 +160,47 @@ public int getRequestPayloadSize() {
         return requestPayloadSize;
     }
 
+    /**
+     * Upload a part of the file by reading a chunk from the InputStream and writing
+     * it to the HTTP request's body. If the number of available bytes is lower than the chunk's
+     * size, all available bytes will be uploaded and nothing more.
+     * The size of the read chunk can be obtained using {@link #getRequestPayloadSize()} and changed
+     * using {@link #setRequestPayloadSize(int)}.
+     * In order to obtain the new offset, use {@link #getOffset()} after this method returns.
+     *
+     * @return Number of bytes read and written.
+     * @throws IOException  Thrown if an exception occurs while reading from the source or writing
+     *                      to the HTTP request.
+     */
+    public int upload() throws IOException, ProtocolException {
+        openConnection(false);
+
+        setChunkSize(bytesRemainingForRequest);
+
+        int bytesRead = input.read(buffer, bytesRemainingForRequest);
+        if(bytesRead == -1) {
+            // No bytes were read since the input stream is empty
+            return -1;
+        }
+
+        // FIXME connection.setRequestProperty("Content-Length", Long.toString(bytesRead));
```

Comment:
What is this FIXME about?

Reply:
It was a reminder before I realized that the Java HTTP library is able to set the Content-Length. I will remove it.